### PR TITLE
[FLINK-17244][docs] Update the Getting Started page

### PR DESCRIPTION
**Pull request by alpinegizmo**
## What is the purpose of the change

* Generally improve the content of this page
* Add the Python Table API code walkthrough
* Add the Hands-on Training

## Translation

* I created FLINK-17504 for updating the Chinese translation

### CODE REVIEW

File: docs/getting-started/index.md [hunk 1]
```diff
@@ -28,34 +28,53 @@ under the License.
...
-  project setup template for Java or Scala and need help setting up
-  dependencies
+* take a look at the **Docker Playgrounds** if you want to see what Flink can do, via a hands-on,
+  docker-based introduction to specific Flink concepts
+* explore one of the **Code Walkthroughs** if you want a quick, end-to-end
+  introduction to one of Flink's APIs
+* work your way through the **Hands-on Training** for a comprehensive,
+  step-by-step introduction to Flink
+* use **Project Setup** if you already know the basics of Flink and want a
+  project template for Java or Scala, or need help setting up the dependencies
```
Comment (NicoK) [inline docs/getting-started/index.md:40]: What do you think about linking these sections directly here? That way, you don't have to look for them in the navigation; especially with the Hands-On Training not being part of this section.

File: docs/getting-started/index.md [hunk 2]
```diff
@@ -28,34 +28,53 @@ under the License.
...
 ### First steps with one of Flink's APIs
 
-The **Code Walkthroughs** are the best way to get started and introduce you step by step to an API.
-A walkthrough provides instructions to bootstrap a small Flink project with a code skeleton and shows how to extend it to a simple application.
+The **Code Walkthroughs** are a great way to get started quickly with a step-by-step introduction to
+one of Flink's APIs. Each walkthrough provides instructions for bootstrapping a small skeleton
+project, and then shows how to extend it to a simple application.
 
-* The [**DataStream API**](./walkthroughs/datastream_api.html) code walkthrough shows how to implement a simple DataStream application and how to extend it to be stateful and use timers. The DataStream API is Flink's main abstraction to implement stateful streaming applications with sophisticated time semantics in Java or Scala.
+* The [**DataStream API**]({% link getting-started/walkthroughs/datastream_api.md %}) code walkthrough shows how
```
Comment (NicoK) [inline docs/getting-started/index.md:58]: Should you actually also include "code walkthrough" in the link? (we're not linking to the DataStream API docs)
```suggestion
* The [**DataStream API**  code walkthrough]({% link getting-started/walkthroughs/datastream_api.md %}) shows how
```

File: docs/getting-started/index.md (hunk 1 above)
Comment (alpinegizmo) [reply docs/getting-started/index.md:40]: I think it might be odd to only link to the Training here, and the other 3 aren't linkable.

Comment (NicoK) [reply docs/getting-started/index.md:40]: true - didn't see that the other 3 are not linkable...

**Pull request merged**